backend/index.ts

```diff
@@ -1,5 +1,40 @@
-import type { _PublicKeyAuthenticatorProtocol } from '../types/interface'
-export interface CreateAuthenticatorOptions {}
-export function createPublicKeyAuthenticator(opts: CreateAuthenticatorOptions): _PublicKeyAuthenticatorProtocol {
-    return {}
+import type { PublicKeyAuthenticatorProtocol } from '../types/interface'
+
+export interface NormalizedCreateOptions {
+    keys: CryptoKeyPair
+    timeout: number
+    rpID: string
+    challenge: BufferSource
+    crossOrigin: boolean
+}
+
+export interface CreateAuthenticatorOptions {
+    getNormalizedCreateOptions(): Promise<NormalizedCreateOptions>
+
+    // sign count
+    getSignCount(key: CryptoKey): Promise<number>
+
+    incrementSignCount(key: CryptoKey): Promise<void>
+
+    hasCredential(options: PublicKeyCredentialCreationOptions | PublicKeyCredentialRequestOptions): Promise<boolean>
```

Comment:
The parameter should be relying party ID and credential ID

Reply:
It's already in the options.

Reply:
Only pass the minimal information it needed.